external-crates/move/move-compiler/src/typing/translate.rs

```diff
@@ -1973,13 +1988,38 @@ fn exp_dotted_to_borrow(
                     (tyloc, "Immutable because of this position"),
                 ))
             }
+            // mark field as used if it's being borrowed
+            if let Some(struct_name) = struct_name(context, &lhs_borrow.ty) {
```

Comment:
This type might be a type variable, so you have to call `unfold_type` first before introspecting